components/eam/src/physics/cam/gw_drag.F90

```diff
@@ -1203,6 +1212,23 @@ subroutine gw_spec_outflds(prefix, lchnk, ncol, ngwv, c, u, v, xv, yv, &
      call outfld(dumc1x,dummyx,ncol,lchnk)
      call outfld(dumc1y,dummyy,ncol,lchnk)
 
+     if (hist_fld_active(trim(dumc1x)//'_100mb')) then
+        call vertinterp(ncol, pcols, pver, state%pmid, 10000._r8, dummyx, dummmy_p_surf)
```

Comment:
Do have a problem here. vertinterp expects `pmid, dummyx`, and `dummy_p_surf` to have the same 1st dimsize. `dummyx `could have smaller 1st dimsize (i.e., ncol < pcols). In that case the shape will be mismatched for the 2nd to last argument in the call. 

The simplest is to also declare `dummyx `as `(pcols,pver)`; and explicitly set upper bound in the following statement
`dummyx(:ncol,:) = taux(:,l,:)`

Reply:
ah, good eye. I like your suggestion to change dummyx and dummyy to have dimension pcols.